### ISSUE TITLE
Stored procedure call doesn't work with named parameter on Postgres

### ISSUE THREAD

**Issue by andreysubbotin** [eclipse-ee4j/eclipselink#623]
EclipseLink version: 2.7.5
Postgres driver version: 42.2.8
Perhaps #362 causes the problem, because it works on 2.7.3 version
```
StoredProcedureQuery query = jpaEm.createStoredProcedureQuery("get_colours", Colour.class);
query.registerStoredProcedureParameter("name_filter", String.class, ParameterMode.IN);
query.setParameter("name_filter", "StoredProcedureTest%");
List<Colour> list = query.getResultList();
```
Exception:
```
Caused by: java.sql.SQLFeatureNotSupportedException: Метод org.postgresql.jdbc.PgCallableStatement.setString(String,String) ещё не реализован
	at org.postgresql.Driver.notImplemented(Driver.java:692)
	at org.postgresql.jdbc.PgCallableStatement.setString(PgCallableStatement.java:753)
	at org.apache.commons.dbcp2.DelegatingCallableStatement.setString(DelegatingCallableStatement.java:1314)
	at org.apache.commons.dbcp2.DelegatingCallableStatement.setString(DelegatingCallableStatement.java:1314)
	at org.eclipse.persistence.internal.databaseaccess.DatabasePlatform.setParameterValueInDatabaseCall(DatabasePlatform.java:2671)
	at org.eclipse.persistence.queries.StoredProcedureCall.prepareStatement(StoredProcedureCall.java:893)
	at org.eclipse.persistence.internal.databaseaccess.DatabaseAccessor.basicExecuteCall(DatabaseAccessor.java:638)
```

**Comment by lukasj** [comment]
@dazey3 any idea about this?

**Comment by dazey3** [comment]
Based on the code for PgCallableStatement:
https://github.com/pgjdbc/pgjdbc/blob/9723f77eefd45d5f7c784c18214baca6e330ae7e/pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java#L752-L754

It would appear that this driver doesn't support java.sql.CallableStatement.setXXX(String parameterName, XXX parameterType). I feel like the driver really should implement this. EclipseLink is just calling java.jql.CallableStatement API. If the driver doesn't support it, we will need to investigate some enhancement in EclipseLink to let the DatabasePlatform (in this case PostgreSQLPlatform) dictate if the platform supports Named Parameters... maybe issue our own warning? I'm not quite sure here.

This stuff is tough for me cause I personally feel like the onus is on the driver implementation and EclipseLink is an innocent bystander. EclipseLink is calling the correct java.sql API it is supposed to. If the driver has issues with that... then they should fix it and users should be aware of the limitations of their platform. But I completely understand the loss of functionality and the regression considering EclipseLink used to translate named -> index for you.

**Comment by rfelcman** [comment]
See my last comment there https://bugs.eclipse.org/bugs/show_bug.cgi?id=559613

**Comment by rfelcman** [comment]
Fix for this bug was merged to master branch (3.0) and 2.7.7 branch/version. I am closing this bug now. If You will discover another trouble related with this bug, feel free to reopen it.
Nightly build with this fix is available at https://www.eclipse.org/eclipselink/downloads/nightly.php .

See PRs
https://github.com/eclipse-ee4j/eclipselink/pull/725
https://github.com/eclipse-ee4j/eclipselink/pull/736
